plugins/module_utils/proxmox.py

```diff
@@ -95,3 +100,43 @@ def _connect(self):
             return ProxmoxAPI(api_host, verify_ssl=validate_certs, **auth_args)
         except Exception as e:
             self.module.fail_json(msg='%s' % e, exception=traceback.format_exc())
+
+    def version(self):
+        apireturn = self.proxmox_api.version.get()
+        return LooseVersion(apireturn['version'])
+
+    def get_node(self, node):
+        nodes = [n for n in self.proxmox_api.nodes.get() if n['node'] == node]
+        return nodes[0] if nodes else None
+
+    def get_nextvmid(self):
+        try:
+            vmid = self.proxmox_api.cluster.nextid.get()
+            return vmid
+        except Exception as e:
+            self.module.fail_json(msg="Unable to get next vmid. Failed with exception: %s" % to_native(e),
+                                  exception=traceback.format_exc())
+
+    def get_vmid(self, name, ignore_missing=False):
+        vms = [vm['vmid'] for vm in self.proxmox_api.cluster.resources.get(type='vm') if vm.get('name') == name]
+
+        if not vms:
+            if ignore_missing:
+                return None
+
+            self.module.fail_json(msg='No VM found with name: %s' % name)
+        elif len(vms) > 1:
+            self.module.fail_json(msg='Multiple VMs found with name: %s, provide vmid instead' % name)
```

Comment:
This could potentially break playbooks that assume that the first vmid match is used, since `proxmox.py` did not have this extra check. Is that something to worry about?

Reply:
I'd be worried more about choosing the wrong VM when two VMs with the same name exist.

Reply:
In that case you should deprecate the behavior, instead of changing it. Breaking changes will not get merged (without a long deprecation period).

Reply:
I added a deprecation message in the cases where the behaviour changed instead of failing.

Reply:
The behavior must still not change. Also you need a `deprecated_features` changelog fragment to document the deprecated behavior.